### PR TITLE
removing block of code that resets 'weights' array

### LINKED ISSUE #6185: Reject levels of cv::CascadeClassifier::detectMultiScale are erased when skipping groupRectangles
I call `cv::CascadeClassifier::detectMultiScale` with `minNeighbors==0` in order to skip `groupRectangles`, while also requesting reject level and weights (by calling the right overloaded method).

Unfortunately, `groupRectangles` overrides all reject levels (variable `weights`) to `1` when `minNeighbors<=0`, as seen here: https://github.com/Itseez/opencv/blob/master/modules/objdetect/src/cascadedetect.cpp#L69
Therefore, I cannot get the reject levels **and** skip grouping.
I seem to miss the benefit of such behavior. If the method should return without running, then the output should be equal to the input. Also note that as oppose to `weights`, the `levelWeights` are not being overridden.

(My motivation in using `detectMultiScale` this way is that I want to call `groupRectangles` independently on the results of `detectMultiScale` and count the number of rectangles grouped in each final rectangle. This metric is a good evaluation to the quality of the detection, other than the reject levels and weights.)

### PR DESCRIPTION
**Pull request by gunshi**
resolves #6185

### CODE REVIEW

File: modules/objdetect/src/cascadedetect.cpp [hunk 1]
```diff
@@ -61,13 +61,7 @@ void groupRectangles(std::vector<Rect>& rectList, int groupThreshold, double eps
...
-        if( weights )
-        {
-            size_t i, sz = rectList.size();
-            weights->resize(sz);
-            for( i = 0; i < sz; i++ )
-                (*weights)[i] = 1;
-        }
+
         return;
     }
```
Comment (StevenPuttemans) [inline modules/objdetect/src/cascadedetect.cpp]: If you want to remove that block, then why not completely remove the if loop?

**Comment by StevenPuttemans** [comment]
I am not sure if simply deleting the problem will fix it. Software depending on this will break down now ... I would rather see an explanation why it was implemented this way then try to change the behaviour.

**Comment by gunshi** [comment]
Maybe, yes. But I ran the tests and they passed, and I went through the entire source of cascadedetect.cpp and there was no processing being done once it was being established after the function call, that the user does not want grouping of rectangles, or that there were no rectangles to group.
But yes, if there are other files in other modules that are depending on cascadedetect's code and utilising the weights array in a post-processing step for some reason, while also depending on it being set to an array of 1's even when there is no grouping, then the code should crash.
I'll look further into it.

**Comment by StevenPuttemans** [comment]
I agree that if the code is actually doing nothing valid for any other part of cascadedetect.cpp, that then the overwriting should be gone. 

An idea, is there a possible case where removing this would lead to an uninitialized weight vector? This might be the reason for adding this and then we just need to reposition it or make the if condition different.

**Comment by alalek** [comment]
BTW,
Original commit with these changes is [here](https://github.com/Itseez/opencv_attic/commit/d6a40f3a3462cc9bd848691699ed5440c670f500), but actual reason of this addition is very unclear.

Moreover this "for" code block has not been used [during tests](http://pullrequest.opencv.org/buildbot/export/opencv_releases/master_coverage-lin64-debug/20160309-012921_32/coverage_html/opencv/modules/objdetect/src/cascadedetect.cpp.gcov.html) at this moment

And [documentation](http://docs.opencv.org/master/d5/d54/group__objdetect.html#ga3dba897ade8aa8227edda66508e16ab9&gsc.tab=0) doesn't cover these cases.

**Comment by StevenPuttemans** [comment]
Then I can agree that we could remove this. Maybe we should mark it as `watchable` or something like that, so that, keeping in mind that if errors around these weights pop up, that this fix could be the reason.

**Comment by vpisarev** [comment]
This PR will be recreated because of CI issues

**Pull request closed without merging**